### PR TITLE
Make interface for routing table (#2046)

### PR DESCRIPTION
**Pull request by sindydwns**
Issue address is #2046

### CODE REVIEW

**Review by longfin — COMMENTED** [review]
How about removing redudant comments on the implmentation side(i.e., `RoutingTable`) with [`<inheritdoc/>`](https://docs.microsoft.com/ko-kr/dotnet/csharp/language-reference/xmldoc/recommended-tags#inheritdoc) in this PR too?

**Review by dahlia — DISMISSED** [review]

**Review by dahlia — DISMISSED** [review]

**Review by OnedgeLee — DISMISSED** [review]

**Comment by longfin** [comment]
@sindydwns latest change looks good to me. but sadly conflicting occured after merging #2201. 😢 


could you check it? 🙏

**Review by OnedgeLee — APPROVED** [review]

File: CHANGES.md [hunk 1]
```diff
@@ -88,6 +88,7 @@ To be released.
  -  Added `TotalSupplyStateCompleters<T>` static class.  [[#915], [#2200]]
  -  Added `StateCompleterSet<T>.TotalSupplyStateCompleter` property.
     [[#915], [#2200]]
+ -  Added `IRoutingTable` interface. [[#2046], [#2229]]
```
Comment (tkiapril) [inline CHANGES.md:91]:
```suggestion
 -  Added `IRoutingTable` interface.  [[#2046], [#2229]]
```

**Review by tkiapril — APPROVED** [review]
Please rebase and resolve merge conflict in CHANGES.md 🙏

**Review by longfin — DISMISSED** [review]

**Review by longfin — APPROVED** [review]

**Review by riemannulus — APPROVED** [review]

**Review by OnedgeLee — DISMISSED** [review]

**Comment by OnedgeLee** [comment]
Seems like rebase on maintainer side would be better, since main branch is frequently updated recently

**Review by tkiapril — APPROVED** [review]

File: CHANGES.md [hunk 2]
```diff
@@ -154,6 +154,7 @@ Released on August 26, 2022.
  -  Added `TotalSupplyStateCompleters<T>` static class.  [[#915], [#2200]]
  -  Added `StateCompleterSet<T>.TotalSupplyStateCompleter` property.
     [[#915], [#2200]]
+ -  Added `IRoutingTable` interface.  [[#2046], [#2229]]
```
Comment (tkiapril) [inline CHANGES.md:157]: Please move your changelog to To be released section.

File: CHANGES.md [hunk 3]
```diff
@@ -207,6 +208,7 @@ Released on August 26, 2022.
 
 [#915]: https://github.com/planetarium/libplanet/issues/915
 [#1538]: https://github.com/planetarium/libplanet/issues/1538
+[#2046]: https://github.com/planetarium/libplanet/issues/2046
```
Comment (tkiapril) [inline CHANGES.md:211]: Please move your changelog to To be released section.

File: CHANGES.md [hunk 4]
```diff
@@ -216,6 +218,7 @@ Released on August 26, 2022.
 [#2219]: https://github.com/planetarium/libplanet/pull/2219
 [#2227]: https://github.com/planetarium/libplanet/pull/2227
 [#2228]: https://github.com/planetarium/libplanet/pull/2228
+[#2229]: https://github.com/planetarium/libplanet/pull/2229
```
Comment (tkiapril) [inline CHANGES.md:221]: Please move your changelog to To be released section.

**Review by tkiapril — CHANGES_REQUESTED** [review]

**Review by tkiapril — DISMISSED** [review]

File: CHANGES.md (hunk 4 above)
Comment (sindydwns) [reply CHANGES.md]: Thanks for letting me know. i move v0.41.0 to v0.42.0.

**Review by tkiapril — DISMISSED** [review]

File: CHANGES.md [hunk 5]
```diff
@@ -20,6 +20,8 @@ To be released.
 
 ### Added APIs
 
+ -  Added `IRoutingTable` interface.  [[#2046], [#2229]]
```
Comment (limebell) [inline CHANGES.md:23]:
```suggestion
 -  Added `IRoutingTable` interface.  [[#2046], [#2229]]
 -  `RoutingTable` became to implement `IRoutingTable` interface.  [[#2046], [#2229]]
```

File: CHANGES.md [hunk 6]
```diff
@@ -20,6 +20,9 @@ To be released.
...
+ -  `RoutingTable` became to implement `IRoutingTable` interface.  [[#2046], [#2229]]
```
Comment (limebell) [inline CHANGES.md:24]:
```suggestion
 -  `RoutingTable` became to implement `IRoutingTable` interface.
    [[#2046], [#2229]]
```

**Review by tkiapril — APPROVED** [review]

**Pull request merged**